google/datalab/contrib/pipeline/_pipeline.py

```diff
@@ -182,48 +163,123 @@ def _get_dependency_definition(task_id, dependencies):
+      return Pipeline._get_bq_load_params(operator_task_details)
+    return operator_task_details
 
   @staticmethod
-  def _add_default_override_params(task_details, operator_type):
-    """ Internal helper that overrides the defaults of an Airflow operator's
-      parameters, when necessary.
-    """
-    if operator_type == 'bq':
-      bq_defaults = {}
-      bq_defaults['use_legacy_sql'] = False
-      for param_name, param_value in bq_defaults.items():
-          if param_name not in task_details:
-            task_details[param_name] = param_value
+  def _get_bq_execute_params(operator_task_details):
+    if 'query' in operator_task_details:
+      operator_task_details['bql'] = operator_task_details['query'].sql
+      del operator_task_details['query']
+
+    # Add over-rides of Airflow defaults here.
+    if 'use_legacy_sql' not in operator_task_details:
+      operator_task_details['use_legacy_sql'] = False
+
+    return operator_task_details
+
+  @staticmethod
+  def _get_bq_extract_params(operator_task_details):
+    if 'table' in operator_task_details:
+      table = google.datalab.bigquery.commands._bigquery._get_table(operator_task_details['table'])
+      operator_task_details['source_project_dataset_table'] = table.full_name
+      del operator_task_details['table']
+    if 'path' in operator_task_details:
+      operator_task_details['destination_cloud_storage_uris'] = '[{0}]'.format(
+          operator_task_details['path'])
+      del operator_task_details['path']
+    if 'format' in operator_task_details:
+      operator_task_details['export_format'] = 'CSV' if operator_task_details['format'] == 'csv' \
+        else 'NEWLINE_DELIMITED_JSON'
+      del operator_task_details['format']
+    if 'delimiter' in operator_task_details:
+      operator_task_details['field_delimiter'] = operator_task_details['delimiter']
+      del operator_task_details['delimiter']
+    if 'compress' in operator_task_details:
+      operator_task_details['compression'] = 'GZIP' if operator_task_details['compress'] else 'NONE'
+      del operator_task_details['compress']
+    if 'header' in operator_task_details:
+      operator_task_details['print_header'] = operator_task_details['header']
+      del operator_task_details['header']
+
+    return operator_task_details
+
+  @staticmethod
+  def _get_bq_load_params(operator_task_details):
+    if 'table' in operator_task_details:
+      table = google.datalab.bigquery.commands._bigquery._get_table(operator_task_details['table'])
+      if not table:
+        table = google.datalab.bigquery.Table(operator_task_details['table'])
+        # TODO(rajivpb): Ensure that mode == create here.
+      operator_task_details['destination_project_dataset_table'] = table.full_name
+      del operator_task_details['table']
+
+    if 'format' in operator_task_details:
+      operator_task_details['export_format'] = 'CSV' if operator_task_details['format'] == 'csv' \
```

Comment:
Google style doesn't allow for backslash line continuation and instead requires implicit line joining by wrapping things in parens.  We don't always follow Google style but it's a nice-to-have.

Reply:
Acknowledged; will keep in mind and retro-fix opportunistically.